### ISSUE TITLE
Which version of bash is mpibash v1.2 tested against?

### ISSUE THREAD

**Issue by ickc** [lanl/MPI-Bash#6]
Hi,

I have some problem to make mpibash works on NERSC (Cori I specifically). Here is the kind of error I encountered:

```bash
Sun Oct 29 15:55:54 2017: [PE_3]:_pmi_alps_get_apid:alps response not OKAY
Sun Oct 29 15:55:54 2017: [PE_3]:_pmi_alps_get_nidlist:pmi_alps_get_apid returned with error: Success
Sun Oct 29 15:55:54 2017: [PE_3]:_pmi_set_nid_list:_pmi_alps_get_nid_list failed
Sun Oct 29 15:55:54 2017: [PE_3]:_pmi_init:_pmi_set_nid_list failed -1
........................................................................
[Sun Oct 29 15:55:54 2017] [c2-0c1s2n1] Fatal error in MPI_Init: Other MPI error, error stack:
MPIR_Init_thread(537): 
MPID_Init(247).......: channel initialization failed
MPID_Init(636).......:  PMI2 init failed: 1 
........................................................................
[Sun Oct 29 15:55:54 2017] [c2-0c1s2n1] Fatal error in MPI_Init: Other MPI error, error stack:
MPIR_Init_thread(537): 
MPID_Init(247).......: channel initialization failed
MPID_Init(636).......:  PMI2 init failed: 1 
srun: error: nid00457: tasks 0-4,7-11,13-31: Aborted
srun: Terminating job step 7914980.0
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 7914980.0 ON nid00457 CANCELLED AT 2017-10-29T15:55:55 ***
srun: error: nid00457: tasks 5-6,12: Killed
```

The `...` are to shorten the message.

I tried these combinations:

compiler | bash version | mpi version | result
-- | -- | -- | --
GNU | 4.4.12 | 1.2 | Failed
Cray | 4.4.12 | 1.2 | Failed
GNU | 4.4.12 | master | Failed
Cray | 4.4.12 | master | Failed
GNU | 4.3.30 | master | Failed
Cray | 4.3.30 | master | Failed

The few things I'd wonder where it might went wrong:

1. bash version
2. `mpicc`. On NERSC, they have 3 major compiler choice, the GNU, Cray, and Intel's. The `cc` command is already a "wrapper" that is sort of like `mpicc`. So I think I'm doing the correct thing here.
3. running the job:
    - I use the example `testmpi.sh`
    - I submitted a batch job script, and that script uses `srun` (sort of like `mpirun`) to call `testmpi.sh` with a specified amount of processes. So I think I'm doing the correct thing here.

Do you have any clue what might went wrong?

Thanks.

**Comment by spakin** [comment]
> Do you have any clue what might went wrong?

Only the vague notion that CLE doesn't like it when you `srun` a script that runs an executable that loads a shared object that initializes MPI.

But you're in luck: I have an account on Cori and was able to fight with this a bit.  At first, I was able to reproduce the error you're seeing.  Then, based on the previous vague notion, tried simplifying the process a bit for CLE's sake and was able to make `testmpi` work.  The trick is to bypass the `mpibash` script and `srun` the `bash` executable directly with both `LD_PRELOAD` and `LD_LIBRARY_PATH` set:

    srun -n 4 --export=LD_PRELOAD=$HOME/mpibash/libexec/mpibash/mpibash.so --export=LD_LIBRARY_PATH=$HOME/mpibash/libexec/mpibash:$LD_LIBRARY_PATH ~/bash-4.3.30/bash $HOME/mpibash/share/doc/mpibash/examples/testmpi

Let me know if that works for you.

— Scott

P.S.  I tested this with `bash` 4.3.30 and the `PrgEnv-gnu/6.0.4` module loaded.
